### PR TITLE
Enabling Touch Events

### PR DESCRIPTION
**Pull request by digitarald**
1st take on fixing #166

touch.js from http://mxr.mozilla.org/mozilla-esr17/source/b2g/chrome/content/touch.js?rev=df7835caf208

Testable on
- http://paulirish.com/demo/multi
- http://openlayers.org/dev/examples/multitouch.html
- http://phoboslab.org/crap/mt.html
- google "test multi touch"

### CODE REVIEW

File: prosthesis/content/touch.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (mykmelez) [inline prosthesis/content/touch.js:1]: Nit: this needs a license header.

File: prosthesis/content/touch.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  // If this is set to true all mouse events will be cancelled by calling
+  // both evt.preventDefault() and evt.stopPropagation().
+  // This will not prevent a contextmenu event to be fired.
+  // This can be turned on if canPreventMouseEvents is true and the consumer
+  // application call evt.preventDefault();
+  let preventMouseEvents = false;
+
+  let TouchEventHandler = {
+    events: ['mousedown', 'mousemove', 'mouseup', 'click', 'unload'],
+    start: function teh_start() {
```
Comment (mykmelez) [inline prosthesis/content/touch.js:32]: I have a hard time not reading this as lolspeak! ;-)

File: prosthesis/content/touch.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            }
+
+            debug('click: fire');
+          }
+          return;
+      }
+
+      let target = eventTarget || this.target;
+      if (target && type) {
+        console.log(type);
```
Comment (mykmelez) [inline prosthesis/content/touch.js:131]: This _console.log()_ call and the one on line 179 should be debug calls instead, as they overwhelm the Error Console with messages. The one on line 135, on the other hand, should be a Cu.reportError().

**Comment by mykmelez** [comment]
> Testable on
> 
> ```
> http://paulirish.com/demo/multi
> http://openlayers.org/dev/examples/multitouch.html
> http://phoboslab.org/crap/mt.html
> ```

They work!

And I see some improvement in Settings. If I mousedown and then drag really slowly for some short period of time, I get into a mode in which I can drag ad infinitum, including quickly.

But if I mousedown and then drag quickly (or even at just a moderate pace), then I see the same problem: the app (or the platform?) starts interpreting the drag as a mouse drag-drop on an image or selection.

And there's something weird going on in Windows 8. There, mousedown and drag on the homescreen accelerates rapidly at some point, no matter how slowly I drag. So I'll drag part of the next page in view and then whoosh! The page scoots the rest of the way in a moment.

So there's some funkiness here. Perhaps the simulation doesn't capture and suppress all mouse events before the platform takes a crack at them, which is why something interprets a rapid movement in the Settings app as a click-drag? And maybe Windows 8 messes with whatever calculation the simulation does to determine how fast to say the finger is moving? (I'm obviously speculating here.)

**Comment by ochameau** [comment]
PR #510 should fix this issue by sharing touch event shim shipped in gaia.

**Comment by mykmelez** [comment]
Fixed by #510!

**Pull request closed without merging**